config.go

```diff
@@ -70,32 +70,57 @@ func (c LoggingConfig) extractor() observability.ContextExtractor {
 }
 
 // MetricsConfig describes how telemetry should be configured.
+// Scope and Tally are exclusive; choose one.
+// If Tally is present, the dispatcher will push metrics to Tally.
+// Otherwise, it is the owner of the dispatcher's repsonsibility to publish
+// metrics.
 type MetricsConfig struct {
+	// Metrics is a *"go.uber.org/net/metrics".Scope for recording stats.
+	// YARPC does not push these metrics; pushing metrics from the root is an
+	// external concern.
+	Metrics *metrics.Scope
 	// Tally scope used for pushing to M3 or StatsD-based systems. By
 	// default, metrics are collected in memory but not pushed.
 	Tally tally.Scope
 }
 
-func (c MetricsConfig) registry(name string, logger *zap.Logger) (*pally.Registry, context.CancelFunc) {
-	r := pally.NewRegistry(
-		pally.Labeled(pally.Labels{
-			"component":  _packageName,
-			"dispatcher": pally.ScrubLabelValue(name),
-		}),
-		// Also expose all YARPC metrics via the default Prometheus registry.
-		pally.Federated(prometheus.DefaultRegisterer),
-	)
+func (c MetricsConfig) scope(name string, logger *zap.Logger) (*metrics.Scope, context.CancelFunc) {
+	var root *metrics.Root
+	var parent *metrics.Scope
+
+	// c.Metrics, c.Tally
+	// yes, yes, panic
+	// yes, nil, use as parent, no push to tally
+	// nil, yes, create root, push to tally
+	// nil, nil, create root, no push to tally
+	if c.Metrics != nil {
+		if c.Tally != nil {
+			panic("yarpc.NewDispatcher expects only one of Metrics.Tally or Metrics.Scope")
```

Comment:
This is a non-critical error - can we instead either return an error and fail dispatcher construction *or* log an error but use `net/metrics` instead of tally?

Reply:
This is what would happen if a noop metrics.Scope was passed in right?